### ISSUE TITLE
Event binding

### ISSUE THREAD

**Issue by JuniorSuperTux** [stsewd/spotify.nvim#7]
Events that gets notified when song updates. This way I can integrate it with things like notify-nvim

**Comment by stsewd** [comment]
Hi, I think I can use the new notify API to print the messages instead of printing to stdout, if that's what you mean.

**Comment by JuniorSuperTux** [comment]
Not exactly, but I'm not exactly sure if my concept is feasible because this plugin is a remote plugin. My idea is that I can pass a lua function to this plugin, and those functions would get called depending on the situation. So it's basically an event listener that gets triggered like say when it's playing a new song. And the reason for this is that I want to implement something that can get the context of every time a new song has been played, and notify the user through  `vim.notify`

**Comment by JuniorSuperTux** [comment]
Or is it feasible to migrate the dbus thing completely to lua?

**Comment by stsewd** [comment]
It's possible to have hooks (and can be lua functions, no need to migrate the plugin for that). But I think you want to trigger those hooks every time a song changes, independently if it was changed using the plugin or not (i.e `:Spotify next`), and that currently isn't implemented.

**Comment by stsewd** [comment]
> every time a song changes, independently if it was changed using the plugin or not 

this should be possible by subscribing to this signal https://specifications.freedesktop.org/mpris-spec/latest/Player_Interface.html#Signal:Seeked, checked and spotify emits that signal when a song changes, and also when is seeked.

We only need to figure out how to keep the pydbus event loop running, and without blocking neovim
